### ISSUE TITLE
Clarify usage of Timeout without asynchronous invocation

### ISSUE THREAD

**Issue by OndrejM** [eclipse/microprofile-fault-tolerance#129]
Usage of `@Timeout` in synchronous mode isn't very clear from Javadocs and the spec - especially all the consequences.

The TCK tests in TimeoutTest expect that the implementation will interrupt the thread when timeout is reached. However, this isn't specified nowhere in Javadocs or PDF spec. Moreover, interrupting a thread only works in some scenarios. It doesn't work if:
 - the thread is blocked on blocking I/O (database, file read/write), an exception is thrown only in case of waiting for a NIO channel
 - the thread isn't waiting (CPU intensive task) and isn't checking for being interrupted
 - the thread will catch the interrupted exception (with a general catch block) and will just continue processing, ignoring the interrupt

In the above cases (the first one more likely than the others), it's technically impossible to suspend execution and throw a timeout exception after the specified timeout. It should be clarified that these limitations apply and that the execution of the operation may, in the worst case, complete in as if the timeout wasn't specified, throwing an exception after it completed if it took longer than the timeout.

In the future version, it should be reconsidered whether in such a case, when processing completes completely but misses the timeout, an exception should be thrown or processing should continue without any notice.

**Comment by OndrejM** [comment]
Ideally, the spec should clarify that the timeout for all I/O operations managed by the MicroProfile runtime should be adjusted according the value of the timeout. E.g. if the annotated method requests a REST resource via JAX-RS, the timeout for the GET operation should be always less or equal than the timeout on the method.

**Comment by Emily-Jiang** [comment]
@OndrejM What you said is completely true. Would you like to submit a PR? If not, I will document this in Javadoc tomorrow.

**Comment by johnament** [comment]
The JAX-RS spec has no way to set timeout.  We can't make such a claim.

**Comment by Emily-Jiang** [comment]
JAX-RS client can set timeout though.

**Comment by johnament** [comment]
Where?

**Comment by Emily-Jiang** [comment]
In JAX-RS 2.0 spec 8.2.1. says A timeout value can be specified when suspending a connection to avoid waiting for a response indefinitely. This is server side. It is orthogonal to our spec as we are on the caller side. I looked at the JAX-RS client api to set timeout. There are some examples to set as a configuration. 
ClientConfig configuration = new ClientConfig();
configuration.property(ClientProperties.CONNECT_TIMEOUT, 1000);
configuration.property(ClientProperties.READ_TIMEOUT, 1000);
Client client = ClientBuilder.newClient(configuration);
I don't know how well used it is. I talked with Ondro and we agree with you to leave JAX-RS bit out. I will create a PR to clarify timeout.
